File: 225.Implement-Stack-using-Queues.js
/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */

class MyStack {
  constructor() {
    this.pushQ = [];
    this.popQ = [];
  }

  push(x) {
    while (this.pushQ.length) {
      this.popQ.push(this.pushQ.pop());
    }

    this.pushQ.push(x);

    while (this.popQ.length) {
      this.pushQ.push(this.popQ.pop());
    }
  }

  pop() {
    while (this.pushQ.length) {
      this.popQ.push(this.pushQ.pop());
    }

    const num = this.popQ.pop();

    while (this.popQ.length) {
      this.pushQ.push(this.popQ.pop());
    }

    return num;
  }

  top() {
    return this.pushQ[0];
  }

  empty() {
    return this.pushQ.length === 0 && this.popQ.length === 0;
  }
}

File: python/225.Implement-Stack-using-Queues.py
class MyStack:

    def __init__(self):
        self.pushQ = []
        self.popQ = []

    def push(self, x: int) -> None:
        while len(self.pushQ):
            self.popQ.append(self.pushQ.pop())

        self.pushQ.append(x)

        while len(self.popQ):
            self.pushQ.append(self.popQ.pop())

    def pop(self) -> int:
        while len(self.pushQ):
            self.popQ.append(self.pushQ.pop())

        num = self.popQ.pop()

        while len(self.popQ):
            self.pushQ.append(self.popQ.pop())

        return num

    def top(self) -> int:
        return self.pushQ[0]

    def empty(self) -> bool:
        return len(self.pushQ) == 0 and len(self.popQ) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()